src/compiler/checker.ts

```diff
@@ -19372,7 +19361,17 @@ namespace ts {
             }
         }
 
+        function checkJsDoc(node: FunctionDeclaration | MethodDeclaration) {
+            if (!node.jsDoc) {
+                return;
+            }
+            for (const doc of node.jsDoc) {
+                checkSourceElement(doc);
+            }
+        }
+
         function checkFunctionOrMethodDeclaration(node: FunctionDeclaration | MethodDeclaration): void {
+            checkJsDoc(node);

```

Comment:
Seems silly to call this in a TS file if the checkers for every possible jsdoc element will test if we're in a TS file and then do nothing?